src/lib.rs

```diff
@@ -2105,6 +2132,23 @@ mod test {
         assert_eq!(it.next(), None);
     }
 
+    #[test]
+    fn is_done_empty() {
+        let mut empty = empty::<u8>();
+        empty.advance();
+        assert!(empty.is_done());
+    }
+
+    #[test]
+    fn is_done_map() {
+        let mut it = convert([1].iter().cloned())
+            .map_ref(|_: &u8| -> &u16 { panic!("only called during get()") });
```

Comment:
It works on 1.19 if you leave the closure to type inference:

```suggestion
            .map_ref::<u16, _>(|_| panic!("only called during get()"));
```

But then you'll find that the `[1]` temporary is dropped, so you also need to make a local for that.